assets/src/edit-story/components/library/panes/media/local/mediaPane.js

```diff
@@ -130,16 +139,36 @@ function MediaPane(props) {
    */
   const onSelect = (mediaPickerEl) => {
     const resource = getResourceFromMediaPicker(mediaPickerEl);
-    // WordPress media picker event, sizes.medium.url is the smallest image
-    insertMediaElement(
-      resource,
-      mediaPickerEl.sizes?.medium?.url || mediaPickerEl.url
-    );
+    try {
+      if (!allowedMimeTypes.includes(resource.mimeType)) {
+        /* translators: %s is a list of allowed file extensions. */
+        const message = sprintf(
+          /* translators: %s: list of allowed file types. */
+          __('Please choose only %s to insert into page.', 'web-stories'),
+          allowedMimeTypes.join(
```

Comment:
Use `allowedFileTypes` from `useConfig` for this instead.

Showing things like `image/png` to users is weird. `allowedFileTypes` gets you `png` etc.